gramps/gen/utils/alive.py

```diff
@@ -272,6 +274,9 @@ def probably_alive_range(self, person, is_spouse=False):
         # ago.
 
         def descendants_too_old (person, years):
+            if person in self.plist:
+                return (None, None, "", None)
+            self.plist.append(person)
```

Comment:
You probably want to add (and check for) just their handle here, not the whole Person object, right? Otherwise you might have a large number of objects in memory.